src/main/java/org/terasology/gookeeper/system/EconomySystem.java

```diff
@@ -79,6 +88,16 @@ public void shutdown() {
 
     @Override
     public void update(float delta) {
+        // Have to add the wallet hud this way since adding in within initialise() results in the economy component to be null
+        if (!setHud) {
+            for (EntityRef wallet : entityManager.getEntitiesWith(EconomyComponent.class)) {
```

Comment:
Is there any other way to do this? This could potentially cause problems in multiplayer, as it would add money to everyone's wallets. It could also cause issues if somehow a player manages to get more than one wallet, as it would give them the entrance fee per wallet (so if they had 2 wallets for some reason, they would give them twice the money)

Reply:
One potential fix I could do is add the wallet block in the GooZoo instead of keeping it with the player.

Reply:
Hmm so on the entrance block or something? (Like we discussed on slack?)
I would say wait for a third opinion. Raise this issue on slack perhaps?

Reply:
Yeah on the entrance block, or maybe add the wallet block within the GooZoo ST.